### PR TITLE
CreateInstance enhancement

### PR DESCRIPTION
**Pull request by songrgg**
* [Feature] Add hostname, project_id, delete_with_instance parameter when creating instance
* [Bug Fix] Fix allocate_public_ip not working whatever the value is.

### CODE REVIEW

**Comment by songrgg** [comment]
@zqfan Please review this.

File: tencentcloud/resource_tc_instance.go [hunk 1]
```diff
@@ -104,6 +104,16 @@ func resourceTencentCloudInstance() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validateInstanceType,
 			},
+			"hostname": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"project_id": &schema.Schema{
```
Comment (zqfan) [inline tencentcloud/resource_tc_instance.go]: project_id can be modified, it should not be ForceNew

File: tencentcloud/resource_tc_instance.go [hunk 2]
```diff
@@ -273,6 +288,18 @@ func resourceTencentCloudInstanceCreate(d *schema.ResourceData, m interface{}) e
...
 	}
+	if hostName, ok := d.GetOk("hostname"); ok {
+		hName := hostName.(string)
+		if len(hName) > 0 {
+			params["HostName"] = hName
+		}
+	}
+	if projectId, ok := d.GetOk("project_id"); ok {
+		pId := projectId.(int)
+		if pId > 0 {
```
Comment (zqfan) [inline tencentcloud/resource_tc_instance.go]: project id can be 0, and it is the default value, leave this check to API side

File: tencentcloud/resource_tc_instance.go (hunk 1 above)
Comment (songrgg) [reply tencentcloud/resource_tc_instance.go]: Sure, fixed.

File: tencentcloud/resource_tc_instance.go (hunk 2 above)
Comment (songrgg) [reply tencentcloud/resource_tc_instance.go]: Fixed, let server validate.

**Comment by likexian** [comment]
Hello @songrgg 
Thank you for your PR, but this repo has been Deprecated, will you please submit your PR to the new repo?
New repo is here: https://github.com/terraform-providers/terraform-provider-tencentcloud

**Comment by likexian** [comment]
# Testing

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudInstance -timeout 120m
?       github.com/terraform-providers/terraform-provider-tencentcloud  [no test files]
?       github.com/terraform-providers/terraform-provider-tencentcloud/gendoc   [no test files]
=== RUN   TestAccTencentCloudInstanceTypesDataSource_basic
--- PASS: TestAccTencentCloudInstanceTypesDataSource_basic (1.53s)
=== RUN   TestAccTencentCloudInstance_basic
--- PASS: TestAccTencentCloudInstance_basic (28.22s)
=== RUN   TestAccTencentCloudInstance_prepaid
--- PASS: TestAccTencentCloudInstance_prepaid (30.18s)
=== RUN   TestAccTencentCloudInstance_sg
--- FAIL: TestAccTencentCloudInstance_sg (33.78s)
    testing.go:599: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.
        
        Error: Error applying: 2 errors occurred:
                * tencentcloud_security_group.my_sg1 (destroy): 1 error occurred:
                * tencentcloud_security_group.my_sg1: (7004)安全组有关联资源，如要删除请先解关联
        
        
                * tencentcloud_security_group.my_sg2 (destroy): 1 error occurred:
                * tencentcloud_security_group.my_sg2: (7004)安全组有关联资源，如要删除请先解关联
        
        
        
        
        
        State: tencentcloud_security_group.my_sg1:
          ID = sg-3o5eda93
          provider = provider.tencentcloud
          description = tf_test_sg_desc
          name = tf_test_sg_name
          project_id = 0
        tencentcloud_security_group.my_sg2:
          ID = sg-7kpmhl71
          provider = provider.tencentcloud
          description = tf_test_sg_desc
          name = tf_test_sg_name
          project_id = 0
=== RUN   TestAccTencentCloudInstance_network
--- PASS: TestAccTencentCloudInstance_network (29.11s)
=== RUN   TestAccTencentCloudInstance_network_no_public_ip
--- PASS: TestAccTencentCloudInstance_network_no_public_ip (29.75s)
=== RUN   TestAccTencentCloudInstance_vpc
--- PASS: TestAccTencentCloudInstance_vpc (45.52s)
=== RUN   TestAccTencentCloudInstance_keypair
--- PASS: TestAccTencentCloudInstance_keypair (281.41s)
=== RUN   TestAccTencentCloudInstance_imageIdChanged
--- PASS: TestAccTencentCloudInstance_imageIdChanged (82.51s)
=== RUN   TestAccTencentCloudInstance_passwordChanged
--- PASS: TestAccTencentCloudInstance_passwordChanged (33.92s)
=== RUN   TestAccTencentCloudInstance_password
--- PASS: TestAccTencentCloudInstance_password (121.27s)
=== RUN   TestAccTencentCloudInstance_hostname
--- PASS: TestAccTencentCloudInstance_hostname (29.46s)
=== RUN   TestAccTencentCloudInstance_projectId
--- PASS: TestAccTencentCloudInstance_projectId (26.80s)

**Comment by likexian** [comment]
Hello @songrgg , thank you for your PR.

**Comment by songrgg** [comment]
Glad to contribute :)

**Pull request merged**